lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3726,21 +3726,32 @@ void ClangImporter::Implementation::lookupAllObjCMembers(
 // deserialized before loading the named member of this class. This allows the
 // decl members table to be warmed up and enables the correct identification of
 // overrides.
-//
-// FIXME: Very low hanging fruit: Loading everything is extremely wasteful. We
-// should be able to just load the name lazy member loading is asking for.
-static void ensureSuperclassMembersAreLoaded(const ClassDecl *CD) {
+static void loadNamedMemberOfSuperclassIfNeeded(const ClassDecl *CD,
+                                                DeclBaseName name) {
   if (!CD)
     return;
 
   CD = CD->getSuperclassDecl();
-  if (!CD || !CD->hasClangNode())
+  if (!CD)
     return;
-  
-  CD->loadAllMembers();
 
-  for (auto *ED : const_cast<ClassDecl *>(CD)->getExtensions())
-    ED->loadAllMembers();
+  auto &ctx = CD->getASTContext();
+
+  if (CD->hasClangNode() && CD->hasLazyMembers()) {
+    auto ci = ctx.getOrCreateLazyIterableContextData(
+        CD, /*lazyLoader=*/nullptr);
+    if (true || !ci->loader->loadNamedMembers(CD, name, ci->memberData))
+      CD->loadAllMembers();
+  }
+
+  for (auto *ED : const_cast<ClassDecl *>(CD)->getExtensions()) {
+    if (ED->hasClangNode() && ED->hasLazyMembers()) {
+      auto ci = ctx.getOrCreateLazyIterableContextData(
+          ED, /*lazyLoader=*/nullptr);
+      if (true || !ci->loader->loadNamedMembers(ED, name, ci->memberData))
```

Comment:
Neat debugging trick, but we oughta get these out before this is merged.

Reply:
Oh, that wasn't supposed to be in there. It works without the 'true ||' too.